### PR TITLE
New command: qmk lint

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

Adds a command for checking over a keyboard/keymap.

## Types of Changes

- [x] Core
- [x] New feature

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by skullydazed** [comment]
I accidentally based this on the branch for #10609. When that's merged I'll retarget master here.

**Comment by skullydazed** [comment]
Some examples:

Screen Shot 2020-10-25 at 11 49 15

**Comment by skullydazed** [comment]
I'm happy with the state of the PR and am ready for reviews.

File: docs/cli_commands.md [hunk 1]
```diff
@@ -178,6 +178,24 @@ Creates a keymap.json from a keymap.c.
...
+qmk lint [-km KEYMAP] [-kb KEYBOARD] [--strict]
+```
+
+This command is directory aware. It will automatically fill in KEYBOARD and/or KEYMAP if you are in a keyboard or keymap directory.
+
+**Examples**:
+
+Do a basic lint check:
+
+    qmk info -kb rominronin/katana60/rev2
```
Comment (fauxpark) [inline docs/cli_commands.md:197]:
```suggestion
    qmk lint -kb rominronin/katana60/rev2
```

File: docs/hardware_keyboard_guidelines.md [hunk 2]
```diff
@@ -3,6 +3,25 @@
 Since starting, QMK has grown by leaps and bounds thanks to people like you who contribute to creating and maintaining our community keyboards. As we've grown we've discovered some patterns that work well, and ask that you conform to them to make it easier for other people to benefit from your hard work.
 
 
+## Use QMK Lint
+
+We have provided a tools, `qmk lint`, which will let you check over your keyboard for problems. We suggest using it frequently while working on your keyboard and keymap.
```
Comment (fauxpark) [inline docs/hardware_keyboard_guidelines.md:8]:
```suggestion
We have provided a tool, `qmk lint`, which will let you check over your keyboard for problems. We suggest using it frequently while working on your keyboard and keymap. 
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: docs/cli_commands.md (hunk 1 above)
Comment (skullydazed) [reply docs/cli_commands.md]: Nice catch.

File: lib/python/qmk/cli/lint.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    if cli.config.lint.keymap:
+        keymap_path = locate_keymap(cli.config.lint.keyboard, cli.config.lint.keymap)
+
+        if not keymap_path:
+            ok = False
+            cli.log.error("Can't find %s keymap for %s keyboard.", cli.config.lint.keymap, cli.config.lint.keyboard)
+        else:
+            keymap_readme = keymap_path.parent / 'readme.md'
+            if not keymap_readme.exists():
+                cli.log.warning('Missing %s', keymap_readme)
```
Comment (zvecr) [inline lib/python/qmk/cli/lint.py:59]: This pathway doesn't honour `Treat warnings as errors`

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -27,7 +27,7 @@ jobs:
         echo ${{ github.event.pull_request.base.sha }}
         echo '${{ steps.file_changes.outputs.files}}'
 
-    - name: Run qmk info
+    - name: Run qmk lint
```
Comment (zvecr) [inline .github/workflows/lint.yml:30]: cant mark it as a suggestion, but you might also want to change the job name on line 9, to 
```yaml
jobs:
  lint:
`

File: lib/python/qmk/cli/lint.py (hunk 3 above)
Comment (skullydazed) [reply lib/python/qmk/cli/lint.py:59]: Nice catch, thanks!

**Review by fauxpark — APPROVED** [review]

File: lib/python/qmk/cli/lint.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    readme_path = keyboard_path / 'readme.md'
+
+    # Check for errors in the info.json
+    if keyboard_info['parse_errors']:
+        ok = False
+        cli.log.error('Errors found when generating info.json.')
+
+    if cli.config.lint.strict and keyboard_info['parse_warnings']:
+        ok = False
+        cli.log.error('Warnings found when generating info.json (Strict mode enabled.)')
```
Comment (Erovia) [inline lib/python/qmk/cli/lint.py:42]:
```suggestion
        cli.log.warning('Warnings found when generating info.json (Strict mode enabled.)')
```

**Review by Erovia — APPROVED** [review]
LGTM

**Pull request merged**